src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/TurnoverPolicyServiceImpl.java

```diff
@@ -39,14 +43,52 @@ public TurnoverPolicy getTurnOverPolicy(String transactionId, String companyAcco
     public List<ValidationError> postTurnoverPolicy(String transactionId, String companyAccountsId,
         TurnoverPolicy turnoverPolicy) throws ServiceException {
 
+        List<ValidationError> validationErrors = validateTurnoverPolicyDetails(turnoverPolicy);
+
+        if (!validationErrors.isEmpty()) {
+            return validationErrors;
+        }
+
         AccountingPoliciesApi accountingPoliciesApi =
             accountingPoliciesService.getAccountingPoliciesApi(transactionId, companyAccountsId);
 
         accountingPoliciesTransformer.setTurnoverPolicy(turnoverPolicy, accountingPoliciesApi);
 
-        return accountingPoliciesService
-            .updateAccountingPoliciesApi(transactionId, companyAccountsId,
+        return accountingPoliciesService.updateAccountingPoliciesApi(transactionId, companyAccountsId,
                 accountingPoliciesApi);
+    }
+
+    /**
+     * Validate the turnover policy details' length before calling the api. This string field should
+     * not be empty or contain empty characters.
+     *
+     * @param turnoverPolicy The turnover policy model that needs to be validated
+     * @return A list of validation errors. This will be empty fo
+     */
+    private List<ValidationError> validateTurnoverPolicyDetails(TurnoverPolicy turnoverPolicy) {
+
+        List<ValidationError> validationErrors = new ArrayList<>();
 
+        if (turnoverPolicy.getIsIncludeTurnoverSelected() &&
+            turnoverPolicy.getTurnoverPolicyDetails() != null &&
+            isInValidStringField(turnoverPolicy.getTurnoverPolicyDetails())) {
+
+            ValidationError error = new ValidationError();
+            error.setFieldPath(TURNOVER_POLICY_DETAILS_FIELD_PATH);
+            error.setMessageKey(INVALID_STRING_SIZE_ERROR_MESSAGE);
+            validationErrors.add(error);
+        }
+        return validationErrors;
+    }
+
+    /**
+     * Check if field contains invalid information: empty or contain empty characters.
+     *
+     * @param field the string field that needs to be assessed
+     * @return
+     */
+    private boolean isInValidStringField(String field) {
```

Comment:
should this be Invalid rather that In Valid?

Reply:
Actually - if this isjust checking if the field is empty should it just be named something like IsFieldEmpty?